src/main/java/com/sonymobile/jenkins/plugins/kerberossso/PluginImpl.java

```diff
@@ -127,6 +135,15 @@ private void removeFilter() throws ServletException {
     }
 
     /**
+     * Create and attach the filter.
+     * @throws ServletException Unable to add filter.
+     */
+    private void registerFilter() throws ServletException {
+        this.filter = new KerberosSSOFilter(createConfigMap(), new SpnegoKerberosAuthenticationFactory());
+        PluginServletFilter.addFilter(filter);
```

Comment:
In this method, yes. The calling code is supposed to deal with that by removing the filter first in case it is re-adding.